azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -59,26 +59,20 @@ def __init__(self, wire_client):
         """
         try:
             self._wire_client = wire_client
+            self._history = None
+            self._extensions_goal_state = None  # populated from vmSettings or extensionsConfig
 
-            # These "basic" properties come from the initial request to WireServer's goalstate API
+            # These properties hold the goal state from the WireServer and are initialized by self._fetch_full_wire_server_goal_state()
             self._incarnation = None
             self._role_instance_id = None
             self._role_config_name = None
             self._container_id = None
-
-            # These "extended" properties come from additional HTTP requests to the URIs included in the basic goal state, or to the HostGAPlugin
-            self._extensions_goal_state = None
             self._hosting_env = None
             self._shared_conf = None
             self._certs = None
             self._remote_access = None
 
-            timestamp = create_timestamp()
-            xml_text, xml_doc, incarnation = GoalState._fetch_goal_state(self._wire_client)
-            self._history = GoalStateHistory(timestamp, incarnation)
-
-            self._initialize_basic_properties(xml_doc)
-            self._fetch_extended_goal_state(xml_text, xml_doc)
```

Comment:
These changes are part of the new fix. I merged _initialize_basic_properties and _fetch_extended_goal_state into a single method (_fetch_full_wire_server_goal_state) and invoked that method from update().

I also removed code from __init__() that is already done by update()